File: DataAccessLayer/Entities/ShoppingCart.cs
using System;
using System.Collections.Generic;
using System.ComponentModel.DataAnnotations;
using System.ComponentModel.DataAnnotations.Schema;
using System.Linq;
using System.Text;
using System.Threading.Tasks;

namespace DataAccessLayer.Entities
{
    public class ShoppingCart : BaseEntity
    {
        [Required, DatabaseGenerated(DatabaseGeneratedOption.None)]
        public int ProductId { get; set; }

        [Required]
        public decimal Quantity { get; set; }

        [Required]
        public decimal UnitPrice { get; set; }

        [Required]
        public decimal TotalPrice { get; set; }

        public Product Product { get; set; }
    }
}

File: UI-Angular/src/models/shoppingCart.ts


export class ShoppingCart{

    id!: number;
    productId: number;
    userId: number;
    quantity: number;
    unitPrice: number;
    totalPrice: number;

    constructor(productId: number, quantity: number, 
    unitPrice: number, totalPrice: number, userId: number){
        this.productId = productId;
        this.quantity = quantity;
        this.unitPrice = unitPrice;
        this.totalPrice = totalPrice;
        this.userId = userId;
    }

}